http/relay/relay.go

```diff
@@ -113,10 +129,10 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // RoundTrip implementation.
 func (p *Proxy) RoundTrip(r *http.Request) (*http.Response, error) {
-	// TODO: give up after N attempts
-
 	b := p.config.Proxy.Backoff.Backoff()
 
+	retries := 0
```

Comment:
I think `b` has `b.Attempts` we could utilize for this

Reply:
I also defaulted `maxRetries` to be config.Proxy.Backoff.Attmpets.